edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -257,6 +250,12 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                         throw new InvalidOperationException($"ConfigSource '{configSourceConfig}' not supported.");
                 }
 
+                metricsConfig = new MetricsConfig(experimentalFeatures.EnableMetrics, MetricsListenerConfig.Create(configuration));
+                builder.RegisterModule(new MetricsModule(metricsConfig, iothubHostname, deviceId));
+
+                diagnosticConfig = new DiagnosticConfig(experimentalFeatures.EnableMetricsUpload, storagePath, configuration);
```

Comment:
We should support to allow configuring different storage path for edge Agent and diagnostic.

Reply:
I plan to replace this with rocksdb in a future pr, which will be more configurable